### ISSUE TITLE
Failing to Migrate Tables

### ISSUE THREAD

**Issue by kmuenkel** [codestudiollc/laravel-totem#76]
TL;DR - The Solution:
In the src/Providers/TotemServiceProvider.php class, this code:
    `$this->mergeConfigFrom(
        __DIR__.'/../../config/totem.php',
        'totem'
    );`
needs to precede the try/catch responsible for registering the ConsoleServiceProvider.
And the check for the table within the 'try' needs to read:
    `if (Schema::hasTable(config('totem.table_prefix', '').'tasks')) {`
instead of:
    `if (Schema::hasTable('tasks')) {`

****

Details:

If you happen to have a table called "tasks" that's in no way related to Totem, and you're leveraging the TOTEM_TABLE_PREFIX environment variable, and you haven't yet run the migrations associate with this package, you'll get an error whenever you try to do virtually anything.  It'll be an SQL error siting the fact that it can't find your totem table.

The TotemServiceProvider does have the ability to automatically run the necessary migrations.  The problem is, it sets this action to wait for Laravel to make sure the "migrator" is actually spun up first.  And in the meantime, it moves on to initiating the ConsoleServiceProvider, which expects those tables to already exist.

There is a check designed to prevent this conflict, but it's not behaving quite right.  Whether or not the ConsoleServiceProvider gets registered depends on whether you have a table called "tasks" in your DB schema already.  That check is not respecting the "totem.table_prefix" value that's contingent on the TOTEM_TABLE_PREFIX environment variable.

In addition, if this condition is not met, then I'm not sure the ConsoleServiceProvider will ever get the opportunity to be registered until the app is restarted, after the run that triggered the migrations.

**Comment by qschmick** [comment]
@kmuenkel Updated. We update the `hasTable` check a little while ago but have now moved the `mergeConfigFrom`. Thank you.

**Comment by jmattheson** [comment]
I'm using Laravel 5.7.19 and I already have a tasks table. I tried putting a TOTEM_TABLE_PREFIX=totem_  or TOTEM_TABLE_PREFIX=true entry in  my .env file but still run into errors while migrating. "Cannot declare class CreateTasksTable, because the name is already in use". I tried the fix mentioned here in issue [#76](https://github.com/codestudiohq/laravel-totem/issues/76) Why not use a prefix all the time instead of creating a headache for the many people who think that task is a good name for a model in their project. Still looking for a solution, laravel-totem looks pretty nice but I haven't been able to try it myself.

**Comment by qschmick** [comment]
@jmattheson this is due to the class name in the migration. The package was created before the larger adoption and fix for the prefix was put into place. That would still cause an issue here due to the class name. A quick fix would be to change the migration class name for your migration. And manually fix your migrations for the name update. We also have to try and keep from causing BC issues as we fix minor edge cases like this
